src/Plugin/IdmsBuilderBase.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+namespace Drupal\thunder_print\Plugin;
+
+use Drupal\Component\Plugin\PluginBase;
+use Drupal\Component\Transliteration\TransliterationInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\ProxyClass\File\MimeType\MimeTypeGuesser;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\thunder_print\IDMS;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\File\MimeType\FileBinaryMimeTypeGuesser;
+use Symfony\Component\HttpFoundation\ResponseHeaderBag;
+use Symfony\Component\HttpFoundation\StreamedResponse;
+
+/**
+ * Base class for Idms builder plugins.
+ */
+abstract class IdmsBuilderBase extends PluginBase implements IdmsBuilderInterface, ContainerFactoryPluginInterface {
+
+  protected $entityTypeManager;
```

Comment:
I miss docblocks here and below. Would provide better DX through autocompletion in IDE.